### ISSUE TITLE
gcovr 5.2 - FileNotFoundError when using "-j" with number greater 1

### ISSUE THREAD

**Issue by andioz** [gcovr/gcovr#628]
I recently updated my environment to gcovr 5.2. Now I get errors like:

```
gcovr -j 4 --root bcm/core/project --xml bcm/core/build/host/debug/coverage.xml --xml-pretty --print-summary bcm/core/build/host/debug
```

```
Traceback (most recent call last):
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/workers.py", line 77, in worker
    work(*args, **kwargs)
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/gcov.py", line 317, in process_datafile
    done = run_gcov_and_process_files(
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/gcov.py", line 490, in run_gcov_and_process_files
    process_gcov_data(fname, covdata, abs_filename, options)
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/gcov.py", line 93, in process_gcov_data
    with io.open(
FileNotFoundError: [Errno 2] No such file or directory: '/local/docker/build/ubuntu-20.04/bcm/core/project/basic_string.h##f147827f66bcffdc471691bd7baf4ed8.gcov'
Traceback (most recent call last):
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/workers.py", line 77, in worker
    work(*args, **kwargs)
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/gcov.py", line 317, in process_datafile
    done = run_gcov_and_process_files(
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/gcov.py", line 490, in run_gcov_and_process_files
    process_gcov_data(fname, covdata, abs_filename, options)
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/gcov.py", line 93, in process_gcov_data
    with io.open(
FileNotFoundError: [Errno 2] No such file or directory: '/local/docker/build/ubuntu-20.04/bcm/core/project/basic_string.h##f147827f66bcffdc471691bd7baf4ed8.gcov'
(ERROR) Uncaught EXCEPTION
Traceback (most recent call last):
  File "/local/docker/build/ubuntu-20.04/venv/bin/gcovr", line 8, in <module>
    sys.exit(main())
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/__main__.py", line 308, in main
    collect_coverage_from_gcov(covdata, options)
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/__main__.py", line 367, in collect_coverage_from_gcov
    contexts = pool.wait()
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/workers.py", line 170, in __exit__
    self.wait()
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/workers.py", line 161, in wait
    raise self.exceptions[0][1]
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/__main__.py", line 367, in collect_coverage_from_gcov
    contexts = pool.wait()
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/workers.py", line 161, in wait
    raise self.exceptions[0][1]
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/workers.py", line 77, in worker
    work(*args, **kwargs)
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/gcov.py", line 317, in process_datafile
    done = run_gcov_and_process_files(
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/gcov.py", line 490, in run_gcov_and_process_files
    process_gcov_data(fname, covdata, abs_filename, options)
  File "/local/docker/build/ubuntu-20.04/venv/lib/python3.8/site-packages/gcovr/gcov.py", line 93, in process_gcov_data
    with io.open(
FileNotFoundError: [Errno 2] No such file or directory: '/local/docker/build/ubuntu-20.04/bcm/core/project/basic_string.h##f147827f66bcffdc471691bd7baf4ed8.gcov'
```

I checked for the file after the run, it exists!

Unfortunately, I have not idea how to reduce my huge project for simple reproduction. No time yet...

But, when I run my project with a previous version like 5.1, everything is fine. When I run without `-j 4` or using `-j 1` instead, everything is fine. 

To me, it looks like some kind of race condition - I tried `-k` option, but no difference.

Environment: Linux, running in a docker image "Ubuntu 20.04".

I'm sorry that I'm not able to give more detailed information, but anyway I hope this helps to identify the problem.

BR Andi

**Comment by Spacetown** [comment]
This bug was introduced with version 5.1 and is fixed on the master branch (see #587). A version 5.2 isn't released yet.

**Comment by andioz** [comment]
My fault, it is version 5.1 as you say. No idea why I have this +1 mistake... 
Thanks for pointing out the fix!
